### ISSUE TITLE
[Sigh] Incorrect profile downloaded when using provisioning_name & ignore_profiles_with_different_name 

### ISSUE THREAD

**Issue by batesyuk** [fastlane/fastlane#5580]
### Issue Description

tl;dr
When using the parameter `provisioning_name` in conjunction with `ignore_profiles_with_different_name` in sigh the wrong profile is downloaded

I have a universal application with a tvOS bundle to be uploaded to the iTunes store. As there is no platform setting in sigh yet when running the command it downloads the iOS profile incorrectly. In an attempt to prevent this I implemented the `provisioning_name` and `ignore_profiles_with_different_name` parameters. The results of which was that it downloaded the same profile as it was before (iOS) and just renamed it to the name I passed in. 
My understanding of this parameter is that it should not behave like this, but should instead skip this profile and download the tvOS one that has the matching name in the developer portal (123 TV).

##### Complete output when running fastlane, including the stack trace and command used

Everything is outputted as being correct

```
                      Summary for sigh 1.9.0 
| app_identifier: my.bundle.id
| adhoc: false                 
| username: my@username.com 
| provisioning_name: 123 TV                
| ignore_profiles_with_different_name: true                        
| skip_certificate_verification: true                        
| skip_install: false                       
| development: false                       
| force: false                       
| skip_fetch_profiles: false                       
```

### Configuration Files

Please copy the complete content of your `Fastfile` and any other configuration files you use below:

Fastfile:
```ruby
  private_lane :sign do
    sigh(app_identifier: 'identifier',
         adhoc: false,
         username: 'apple_username',
         provisioning_name: '123 TV',
         ignore_profiles_with_different_name: true,
         skip_certificate_verification: true)
    ENV['PROFILE_UDID'] = lane_context[SharedValues::SIGH_UDID]
  end
```

### Environment

fastlane version (run `fastlane -v`): 1.9.0

Do you use bundler to execute fastlane (i.e. `bundle exec fastlane`)? yes - 1.12.5

Do you use a Ruby environment manager (e.g. `chruby`, `rbenv`, `rvm`)? rvm - 1.27.0

**Comment by batesyuk** [comment]
By bad... this does actually seem to be working as expected.

**Comment by TKBurner** [comment]
@batesyuk No problem! Really glad to hear you got up and running 👍
